Add RPM specifiacation

mkgrgis opened the pull request:
Create `rpm.spec`.
In continue of https://github.com/pgspider/sqlite_fdw/pull/39

Based on
* https://rpm-software-management.github.io/rpm/manual/spec.html
* https://github.com/scj643/sqlite_fdw/tree/master
* `rpm.spec` from https://rhel.pkgs.org/9/postgresql-14-x86_64/sqlite_fdw_14-2.3.0-1.rhel9.x86_64.rpm.html

Thanks to @scj643 and @devrimgunduz

mkgrgis committed:
Add RPM specifiacation

Create rpm.spec
Based on
* https://rpm-software-management.github.io/rpm/manual/spec.html
* https://github.com/scj643/sqlite_fdw/tree/master
* rpm.spec from https://rhel.pkgs.org/9/postgresql-14-x86_64/sqlite_fdw_14-2.3.0-1.rhel9.x86_64.rpm.html

Thanks to https://github.com/scj643 and https://github.com/devrimgunduz

t-kataym commented:
@mkgrgis , Thank you for creating the PR.  
Is it no problem to start the review?

@MinhLA1410 , 
Please review it.

mkgrgis commented:
> @mkgrgis , Thank you for creating the PR. Is it no problem to start the review?

Yes, no problems. I'll study about RPM directive words if there will detected any problems.

MinhLA1410 commented:
Hi @mkgrgis 
I tried to build rpm file from your spec. But it does not work. Could you please check again
```
error: parse error in expression
error: ${HOME}/sqlite_fdw/rpm.spec:25: bad %if condition
```

mkgrgis commented:
Thanks, @MinhLA1410 ! I'll try to get rpm environment, reproduce this problem and check again.

devrimgunduz commented:
Why are you trying to add spec file to sqlite_fdw repo? I'm already maintaining it in upstream repo.

mkgrgis commented:
Hello, @devrimgunduz ! Thanks for reply and for rpms :-) . I try to add spec file because i want to get rpm files for my colleagues with some new features or with my experiments for user testing. Your rpm files with sqlite_fdw likes me because there is good implementation of lifecycle processes.

mkgrgis committed:
Merge branch 'pgspider:master' into draft_rpm_spec

mkgrgis committed:
Merge branch 'pgspider:master' into draft_rpm_spec

mkgrgis committed:
Merge branch 'pgspider:master' into draft_rpm_spec

mkgrgis committed:
Merge branch 'pgspider:master' into draft_rpm_spec

mkgrgis committed:
Merge branch 'pgspider:master' into draft_rpm_spec

mkgrgis committed:
Merge branch 'pgspider:master' into draft_rpm_spec

mkgrgis committed:
Merge branch 'pgspider:master' into draft_rpm_spec

devrimgunduz commented:
The spec file is quite old, does not support RHEL 8, 9 and 10, LLVM dependencies are ancient. Also makes a few false assumptions.

mkgrgis committed:
Update rpm.spec

Some fixes

mkgrgis commented:
> LLVM dependencies are ancient

@devrimgunduz , could you please continue actualization of PRM specification after my https://github.com/pgspider/sqlite_fdw/pull/80/commits/7465d03243f84f6eda2d6ce739516657882ae3ea before successfully package building?

mkgrgis committed:
Update rpm.spec, for "without gis support" variant

Based on confirmed by real RPM creating process file variant from https://github.com/pgsty/rpm/blob/main/rpmbuild/SPECS/sqlite_fdw.spec
